Movies_MVVMTests/DataServiceTests.swift

```diff
@@ -0,0 +1,89 @@
+// DataServiceTests.swift
+// Copyright © Aleksandr Dorofeev. All rights reserved.
+
+import XCTest
+
+@testable import Movies_MVVM
+
+/// Data service tests
+final class DataServiceTests: XCTestCase {
+    // MARK: - Private Constants
+
+    private enum Constants {
+        static let emptyString = ""
+        static let mockCategory = "mockCategory"
+        static let fooData = "Foo"
+        static let barData = "Bar"
+        static let mockMovieId = 2
+        static let mockMovieDetailId = 1
+        static let mockGenres: [MovieDetail.Genres] = []
+    }
+
+    // MARK: - Private properties
+
+    private var mockMovies: [Movie] = [
+        Movie(
+            id: Constants.mockMovieId,
+            posterPath: Constants.barData,
+            title: Constants.barData,
+            voteAverage: 1
+        )
+    ]
+    private var mockMovieDetail = MovieDetail(
+        posterPath: Constants.fooData,
+        title: Constants.fooData,
+        voteAverage: 0,
+        releaseDate: Constants.fooData,
+        genres: Constants.mockGenres,
+        overview: Constants.fooData,
+        id: Constants.mockMovieDetailId
+    )
+    private var dataService: DataServiceProtocol?
+
+    // MARK: - Public methods
+
+    override func setUp() {
+        dataService = DataService()
+    }
+
+    override func tearDown() {
+        dataService = nil
+    }
+
+    func testWriteMovies() {
+        XCTAssertNotNil(dataService?.writeMovieData(movies: mockMovies, category: Constants.mockCategory))
+    }
+
+    func testWriteMovieDetail() {
+        XCTAssertNotNil(
+            dataService?
+                .writeMovieDetailData(movieDetail: mockMovieDetail, id: Constants.mockMovieDetailId)
+        )
+    }
+
+    func testReadMovies() {
+        let result = dataService?.readMovieData(category: Constants.mockCategory)
```

Comment:
```suggestion
        dataService?.writeMovieData(movies: mockMovies, category: Constants.mockCategory)
        let result = dataService?.readMovieData(category: Constants.mockCategory)
```
вот тут закинул и прочитал - нормальный тест, только потом желательно удалить из сервиса, чтобы в следующий раз снова добавлять и проверять

Reply:
Не очень понял о каком списке речь?

Reply:
Я не реализовывал метод удаления данных из базы данных, ради теста это делать считаю не целесообразным

Reply:
Решил протеститровать не реальный объект а мок, в таком случае могу занилить